core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -169,7 +169,6 @@ export class ExplorationSaveAndPublishButtonsComponent
       .finally(() => {
         this.publishIsInProcess = false;
         this.loadingDotsAreShown = false;
-        this.entityTranslationsService.reset();
```

Comment:
Just curious, why are we removing this?

Reply:
Because it resets our translations dict used to keep track of translation changes on simply opening the publish modal. I'm unsure why it was being reset in the first place.